jsinterop-ts-defs-processor/src/main/java/com/vertispan/tsdefs/model/TsVariable.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright © 2023 Vertispan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.vertispan.tsdefs.model;
+
+import static com.vertispan.tsdefs.Formatting.NONE;
+import static com.vertispan.tsdefs.Formatting.resolveName;
+import static java.util.Objects.nonNull;
+
+import com.vertispan.tsdefs.builders.Deprecation;
+import com.vertispan.tsdefs.builders.HasDocs;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public abstract class TsVariable {
+  protected final String name;
+  protected final TsType type;
+  protected TsDoc tsDoc;
+  protected boolean jsOptional;
+  protected boolean deprecated;
+  protected boolean varargs = false;
+  protected List<TsModifier> modifiers = new ArrayList<>();
+
+  protected TsVariable(String name, TsType type) {
+    this.name = name;
+    this.type = type;
+    this.type.setNullUnionType(getNullUnionType());
```

Comment:
This seems a little weird, the variable mutating the type.

It also is slightly dangerous, calling an abstract method from inside a constructor (when fields, even final ones, haven't been initialized).

Not a strong objection, just want to point this out.